### PR TITLE
Update template to Webpack 5 for better experience

### LINKED ISSUE #12: Out of the box error on first DEV run
On a fresh clone, the first ```npm run dev``` results in an error loading the app style sheet.  The console error is something like this: 

```Refused to apply style from '{bundles.css blah blah}' because its MIME type ('text/html') is not a supported stylesheet MIME type, and strict MIME checking is enabled.```

To work-around it, I built the project (which produces a physical *bundles.css* file) once, and resumed devving. That seemed to clear the error.

Leaving this open as it could be that I'm missing something fundamental. If not, good to close it.

### LINKED ISSUE #31: Hot reload, how to watch for global.css too
I'm new to webpack. I opted to use this template, instead of the default one with Rollup, because webpack comes with a proxy feature that I need for development and Rollup currently does not.

The problem is that the `global.css` inside the public folder does not trigger hot reload when changes are made. Making changes to the styles in .svelte files does work as expected. I tested making changes to the same `global.css` in the default template with Rollup and that triggers hot reload just fine. Anyone know which settings I'm supposed to add to `webpack.config.js` to get this to work?

### LINKED ISSUE #34: use same file structure as svelte rollup template
for example, write JS to `public/build/bundle.js` instead of `public/bundle.js`

benefit: this makes easier changing between rollup and webpack

### PR DESCRIPTION
**Pull request by non25**
I find it sad that there's 18k weekly downloads of `svelte-loader` and these users don't get the best possible svelte experience on Webpack. This template doesn't show a good example of how this could be done either, and it is the most important one, because it is official.

This fixes the situation by updating `svelte-loader` from master, which has Webpack 5 and HMR support and also updating everything else to the most recent version. I believe starters don't want to use stale deps.

Issues: Fixes #12, Fixes #31, Fixes #34
PRs: Fixes #33, Fixes #24, Fixes #21, Fixes #28

### CODE REVIEW

**Comment by benmccann** [comment]
looks like a good idea. we will need to wait for a new release of `svelte-loader`

File: package.json [hunk 1]
```diff
@@ -2,19 +2,18 @@
...
-    "svelte-loader": "2.13.3",
-    "webpack": "^4.30.0",
-    "webpack-cli": "^3.3.0",
-    "webpack-dev-server": "^3.3.1"
+    "cross-env": "^7.0.3",
+    "css-loader": "^5.0.1",
+    "mini-css-extract-plugin": "^1.3.4",
+    "style-loader": "^2.0.0",
+    "svelte": "^3.31.2",
+    "svelte-loader": "sveltejs/svelte-loader",
```
Comment (benmccann) [inline package.json]: we should make this `^3.0.0`

**Comment by benmccann** [comment]
The new version of `svelte-loader` has been released now

File: webpack.config.js [hunk 2]
```diff
@@ -51,5 +61,8 @@ module.exports = {
 			filename: '[name].css'
 		})
 	],
-	devtool: prod ? false: 'source-map'
+	devtool: prod ? false: 'source-map',
```
Comment (benmccann) [inline webpack.config.js]:
```suggestion
	devtool: prod ? false : 'source-map',
```

**Review by benmccann — APPROVED** [review]

**Comment by non25** [comment]
I've been thinking about `emitCss: false` for HMR to work and somehow having `bundle.css` not to break reloading. The ways to solve this are:

1. Prebuild `bundle.css` before running devmode. Drawbacks: old css from previous production build interfering with new, need to prebuild stuff
2. Make an empty `bundle.css` in public before running devmode. Drawbacks: kinda manualish or could introduce wierd stuff in `webpack.config.js` to make it automatic
3. Move `global.css` as an import to the `main.js`. Can't see drawbacks for that, allows HMR to work for `global.css`, fixes #31. 

Currently, components don't output css in devmode, and there's no global imported css, so by default there won't be any `bundle.css` in devmode in real/virtual fs, so hot reloading will break.

^ That is also because we are using `style-loader`. Moving `global.css` to src won't fix #12 by itself.
We will also need to use `mini-css-extract-plugin` in devmode, but that is preferable nowadays anyway.

**Comment by non25** [comment]
Also I think #4, #9, #26, #30, #32 are irrelevant now and could be closed.

**Pull request merged**